File: dashboard/frontend/src/types/costs.ts
export interface DailyCostPoint {
  date: string
  cost: number
}

export interface CostByService {
  service_type: string
  total_cost: number
  percentage: number
}

export interface CostByOwner {
  owner: string
  total_cost: number
  percentage: number
}

export interface CostAnomaly {
  date: string
  actual_cost: number
  expected_cost: number
  deviation_pct: number
}

export interface CostSummary {
  total_cost: number
  average_daily: number
  trend_direction: string
  trend_pct: number
  anomaly_count: number
}

File: dashboard/backend/app/schemas/costs.py
from pydantic import BaseModel


class DailyCostPoint(BaseModel):
    date: str
    cost: float
    service_type: str = ""


class CostByService(BaseModel):
    service_type: str
    total_cost: float
    percentage: float


class CostByOwner(BaseModel):
    owner: str
    total_cost: float
    percentage: float


class CostAnomaly(BaseModel):
    date: str
    actual_cost: float
    expected_cost: float
    deviation_pct: float
    service_type: str = ""


class CostSummary(BaseModel):
    total_cost: float
    average_daily: float
    trend_direction: str
    trend_pct: float
    anomaly_count: int
